File: app/src/main/java/com/isp/app/network/ApiService.java
package com.isp.app.network;

import com.isp.app.data.models.Program;
import java.util.ArrayList;
import java.util.List;

public class ApiService {

    public List<Program> getPrograms() {
        // Simulate fetching data from a network or database
        List<Program> programs = new ArrayList<>();
        programs.add(new Program("Computer Science", "CS101"));
        programs.add(new Program("Business Administration", "BA202"));
        programs.add(new Program("Mechanical Engineering", "ME303"));
        return programs;
    }
} 

File: app/src/main/java/com/college/app/network/ApiService.kt
package com.college.app.network

import com.college.app.data.models.Program
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiService {
    @GET("programs")
    suspend fun getPrograms(): List<Program>

    @GET("programs/{programId}/courses")
    suspend fun getCourses(@Path("programId") programId: String): List<Course>
} 